### PR TITLE
elmPackages.elm-land: init at 0.19.4

### PR DESCRIPTION
**Pull request by zupo**
## Description of changes

Support installing https://elm.land/ directly from nixpkgs. 

Thanks @Lassulus for hand-holding!

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by qbit** [comment]
Result of `nixpkgs-review pr 267725` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>elmPackages.create-elm-app</li>
    <li>elmPackages.elm-analyse</li>
    <li>elmPackages.elm-coverage</li>
    <li>elmPackages.elm-land</li>
    <li>elmPackages.elm-pages</li>
    <li>elmPackages.elm-review</li>
    <li>elmPackages.elm-upgrade</li>
    <li>elmPackages.elm-verify-examples</li>
  </ul>
</details>

**Review by qbit — APPROVED** [review]

File: pkgs/development/compilers/elm/default.nix [hunk 1]
```diff
@@ -274,6 +274,19 @@ in lib.makeScope pkgs.newScope (self: with self; {
         }
       );
 
+      elm-land = nodePkgs."elm-land".overrideAttrs (
+        old: {
+          nativeBuildInputs = (old.nativeBuildInputs or []) ++ [ makeWrapper old.nodejs.pkgs.node-gyp-build ];
```
Comment (turboMaCk) [inline pkgs/development/compilers/elm/default.nix:279]: I don't understand why this is necessary.

Comment (zupo) [reply pkgs/development/compilers/elm/default.nix:279]: Bah that's an artifact from copy/pasting the definition from `elm-spa` in the same file. I'll remove it. Thanks for catching it!

**Review by turboMaCk — APPROVED** [review]
LGTM

**Comment by dotlambda** [comment]
Can we use `buildNpmPackage`? See #229475.

**Comment by turboMaCk** [comment]
for some of the things in elmPackages yes -- we would just loose the ability to do semi-automatic updates. For other things, definitely the ones that use bin-wrap package we can't or at least we don't know how. One package was already ported but it's hard to find a motivation since it seems to make maintenance harder and can't be applied universally.

**Comment by dotlambda** [comment]
I don't think it's okay to keep adding uses of the node2nix antipattern.

**Comment by turboMaCk** [comment]
That's fair. I think this package might work with `buildNpmPacage` out of the box but we would need to try it first. I plan to eventually port everything but I need to find time to find suitable solution to problematic parts first so it will take some time.

That's being said the advantage of switching to `buildNpmPackage` for `elmPackages` is not that significant because we generally do want to always update existing and our generation script is independent of the one from `nodePackages`. Ideally we also want to maintain this process with `buildNpmPackage` but I don't think it will be possible by design. The only reason for the switch is consistency with changes in `nodePacakges`.

**Pull request merged**